map/bookmark_helpers.cpp

```diff
@@ -479,7 +479,24 @@ bool SaveKmlFile(kml::FileData & kmlData, std::string const & file, KmlFileType
   try
   {
     FileWriter writer(file);
-    success = SaveKmlData(kmlData, writer, fileType);
+    switch (fileType)
```

Comment:
This code can be simplified, and Save... functions can be removed. It's bad to mix both exceptions and return values in C++.

```
  try
  {
    FileWriter writer(file);
    switch (fileType)
    {
    case KmlFileType::Text:
      kml::SerializerKml{kmlData}.Serialize(writer);
      break;

    case KmlFileType::Binary:
      kml::binary::SerializerKml{kmlData}.Serialize(writer);
      break;

    case KmlFileType::Gpx:
      kml::gpx::SerializerGpx{kmlData}.Serialize(writer);
      break;
    }
    default: CHECK(false, ("Unexpected KmlFileType", fileType, "for", file));
  }
  catch (Writer::Exception const & e)
  {
    LOG(LWARNING, (fileType, "writing failure for", file, e.what()));
    return false;
  }
  catch (std::exception const & e)
  {
    LOG(LWARNING, (fileType, "serialization failure for", file, e.what()));
    return false;
  }
  return true;
```

Reply:
Fixed code and removed SaveGpxData. For SaveKmlData it is not that easy - it is used in unit-tests with custom writer via BookmarkManager::SaveBookmarkCategory. That one is test-only also, but uses some private methods of BookmarkManager, so need to think how to fix it - make them public and move logic of these two functions to test or leave as is.

Reply:
Good point, I missed it somehow. Then maybe simplify the code a bit, and also use
```
return SaveKmlData(kmlData, writer, fileType);
```
as it catches all exceptions already? (a comment about it can be added).

Reply:
Restored SaveGpxData, re-used SaveKmlData, SaveKmlFile reworked to avoid exception handling, only return of result code.